loopy/kernel/instruction.py

```diff
@@ -22,15 +22,54 @@
 
 from sys import intern
 from pytools import ImmutableRecord, memoize_method
+from pytools.tag import Tag, tag_dataclass, Taggable
 from loopy.diagnostic import LoopyError
 from loopy.tools import Optional
 from warnings import warn
 import islpy as isl
 
 
+# {{{ instruction tags
+
+@tag_dataclass
+class LegacyStringInstructionTag(Tag):
```

Comment:
How about we just rename it to StringInstructionTag. I can see that supplying pure string tags can be quite convenient and from the looks of it there are no plans to deprecate pure string-valued tags.

Reply:
> there are no plans to deprecate pure string-valued tags.

Not true. I think string tags are not that useful: they don't allow parameters, and they're subject to name clashes. Language syntax and pattern matching (both subject to being redone) currently rely on string tags (and do not have alternatives), that's why I've not slapped the `LegacyStringInstructionTag` with a deprecation warning. But in general, I think getting rid of string tags is the correct move. In 09bf5f38, I've added a comment to that effect. What do you think?

Reply:
Ping?

Reply:
Yep, that's sounds like a good direction. Thanks for clarifying it!

Reply:
#250